TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/KafkaPrinter.java

```diff
@@ -164,24 +162,49 @@ public static void main(String[] args) {
         properties.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
         properties.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupId);
 
-        Map<Partition, PartitionTracker> capturedRecords = new HashMap<>();
+        Map<TopicPartition, PartitionTracker> capturedRecords = new HashMap<>();
         if (!params.partitionLimits.isEmpty()) {
             for (String partitionLimit : params.partitionLimits) {
                 String[] partitionElements = partitionLimit.split(":");
                 if (partitionElements.length != 3) {
                     throw new ParameterException("Partition limit provided does not match the expected format: topic_name:partition_id:num_records, actual value: " + partitionLimit);
                 }
-                Partition partition = new Partition(partitionElements[0], Integer.parseInt(partitionElements[1]));
+                TopicPartition partition = new TopicPartition(partitionElements[0], Integer.parseInt(partitionElements[1]));
                 if (capturedRecords.containsKey(partition)) {
                     throw new ParameterException("Duplicate parameter limit detected for the partition: " + partition);
                 }
                 capturedRecords.put(partition, new PartitionTracker(0, Long.parseLong(partitionElements[2])));
             }
         }
 
+        String baseOutputPath = params.outputDirectoryPath == null ? "./" : params.outputDirectoryPath;
+        baseOutputPath = !baseOutputPath.endsWith("/") ? baseOutputPath + "/" : baseOutputPath;
+        String uuid = UUID.randomUUID().toString();
+        boolean separatePartitionOutputs = false;
+        Map<Integer, CodedOutputStream> partitionOutputStreams = new HashMap<>();
+        // Grab all partition records
+        if (capturedRecords.isEmpty()) {
+            OutputStream os = params.outputDirectoryPath == null ? System.out : new FileOutputStream(String.format("%s%s_%s_%s.proto", baseOutputPath, params.kafkaTrafficTopic, "all", uuid));
+            partitionOutputStreams.put(0, CodedOutputStream.newInstance(os));
```

Comment:
This block can roll into the next block - excise and make the first if statement include `captureRecords.isEmpty() ||...`